### PR TITLE
[DST-450]: rework design

### PR DESCRIPTION
**Pull request by aromko**
Focus on general layout and compound component as first example. The latest update from state management isn't merged currently. 
Also added the demo preview.

### CODE REVIEW

**Review by tirado-rx — CHANGES_REQUESTED** [review]
Hi Marcel, see my comments on the board for my review. In general, you want to make as few design decisions as necessary before user testing, so I think that most design work will follow after the tests.

If you like, we can also discuss the scope and requirements of the product together in order to decide which recommendations (if any) are most important to implement now.

File: src/components/DemoLink.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Link } from '@tanstack/react-router';
+import { ExternalLink } from '@marigold/icons';
+import { Button } from '@marigold/components';
+
+const DemoLink = ({
+  destination,
+  text = 'View demo',
```
Comment (sebald) [inline src/components/DemoLink.tsx:7]: Why not just use `children` here?

File: src/components/DemoLink.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}: {
+  destination: string;
+  text?: string;
+}) => {
+  return (
+    <div className="w-fit">
+      <Link target="_blank" to={destination} search="">
+        <Button className="flex-1 cursor-pointer rounded-md bg-white p-4 font-black text-gray-800 hover:bg-gray-300">
```
Comment (sebald) [inline src/components/DemoLink.tsx:15]: no puttin buttons into links :D (add new variant?)

**Review by sebald — CHANGES_REQUESTED** [review]

File: src/components/DemoLink.tsx (hunk 2 above)
Comment (aromko) [reply src/components/DemoLink.tsx]: changed it

Comment (aromko) [reply src/components/DemoLink.tsx]: But I don't need a variant yet or?

**Review by sebald — APPROVED** [review]

**Review by sebald — APPROVED** [review]

**Pull request merged**